File: 118. Pascal's Triangle/118. Pascal's Triangle.dart
class Solution {
  List<List<int>> generate(int numRows) {
    List<List<int>> result = [];
    
    for (int i = 0; i < numRows; i++) {
      List<int> row = List<int>.filled(i + 1, 1);
      
      for (int j = 1; j < i; j++) {
        row[j] = result[i-1][j-1] + result[i-1][j];
      }
      
      result.add(row);
    }
    
    return result;
  }
}

File: 118. Pascal's Triangle/118. Pascal's Triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var result = new List<IList<int>>();
        
        for (int i = 0; i < numRows; i++) {
            var row = new List<int>();
            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    row.Add(1);
                } else {
                    row.Add(result[i-1][j-1] + result[i-1][j]);
                }
            }
            result.Add(row);
        }
        
        return result;
    }
}